tests/subsys/canbus/isotp/conformance/testcase.yaml

```diff
@@ -1,7 +1,54 @@
+#
+# There are 4 different test cases that have to be considered:
+#
+# | Case # | Controller type        | Selected mode        |  Example board |
+# +--------+------------------------+----------------------+----------------+
+# |    1   | Classical CAN only     | CONFIG_CAN_FD_MODE=n | nucleo_f072    |
+# |    2   | Classical CAN only     | CONFIG_CAN_FD_MODE=y | nucleo_f072    |
+# |    3   | Classical CAN + CAN-FD | CONFIG_CAN_FD_MODE=n | native_posix   |
+# |    4   | Classical CAN + CAN-FD | CONFIG_CAN_FD_MODE=y | native_posix   |
```

Comment:
These cases significantly increase the time required for running the ISO-TP conformance test suite on real hardware. Can the tests be optimized in time somehow?

Example of time spent running this test suite on five random boards on my desk:
```
INFO    -  6/35 frdm_k64f                 tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_64 PASSED (device: 0240000047784e45004e9003d9170022e561000097969900, 8.151s)
INFO    -  7/35 lpcxpresso55s36           tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_64 PASSED (device: 001069887263, 14.602s)
INFO    -  8/35 stm32f3_disco             tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_64 PASSED (device: 066EFF555052836687113518, 4.123s)
INFO    -  9/35 mimxrt1060_evkb           tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_64 PASSED (device: 0229000033f4cd1100000000000000000000000097969905, 33.479s)
INFO    - 10/35 nucleo_g474re             tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_64 PASSED (device: 0041001B3438510B34313939, 13.539s)
INFO    - 11/35 frdm_k64f                 tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_32 PASSED (device: 0240000047784e45004e9003d9170022e561000097969900, 10.311s)
INFO    - 12/35 lpcxpresso55s36           tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_32 PASSED (device: 001069887263, 11.119s)
INFO    - 13/35 stm32f3_disco             tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_32 PASSED (device: 066EFF555052836687113518, 3.966s)
INFO    - 14/35 mimxrt1060_evkb           tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_32 PASSED (device: 0229000033f4cd1100000000000000000000000097969905, 37.112s)
INFO    - 15/35 nucleo_g474re             tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.txdl_32 PASSED (device: 0041001B3438510B34313939, 13.679s)
INFO    - 16/35 frdm_k64f                 tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.unused PASSED (device: 0240000047784e45004e9003d9170022e561000097969900, 16.917s)
INFO    - 17/35 lpcxpresso55s36           tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.unused PASSED (device: 001069887263, 10.552s)
INFO    - 18/35 stm32f3_disco             tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.unused PASSED (device: 066EFF555052836687113518, 14.584s)
INFO    - 19/35 mimxrt1060_evkb           tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.unused PASSED (device: 0229000033f4cd1100000000000000000000000097969905, 33.130s)
INFO    - 20/35 nucleo_g474re             tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance.fd.unused PASSED (device: 0041001B3438510B34313939, 14.217s)
INFO    - 21/35 frdm_k64f                 tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance PASSED (device: 0240000047784e45004e9003d9170022e561000097969900, 17.851s)
INFO    - 22/35 lpcxpresso55s36           tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance PASSED (device: 001069887263, 11.913s)
INFO    - 23/35 mimxrt1060_evkb           tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance PASSED (device: 0229000033f4cd1100000000000000000000000097969905, 31.035s)
INFO    - 24/35 stm32f3_disco             tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance PASSED (device: 066EFF555052836687113518, 14.670s)
INFO    - 25/35 nucleo_g474re             tests/subsys/canbus/isotp/conformance/canbus.isotp.conformance PASSED (device: 0041001B3438510B34313939, 14.054s)
```

Do we really need to test with different TX_DLs on all boards or could a simple unit-like test (e.g. on POSIX) be sufficient for the different configurations? I mean, these tests validate the protocol implementation, not as much as the protocol interaction with the particular CAN controller?

Another option could be to tag some of these tests as "slow".

Reply:
There will be some more upcoming ISO-TP updates. Let's leave the optimization of test time for a future PR.